src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -275,8 +338,22 @@ public synchronized boolean lock(List<LockableResource> resources,
 		}
 		if (!needToWait) {
 			for (LockableResource r : resources) {
+				LOGGER.log(Level.FINE, "Setting lock for: " + r.getUid());
 				r.unqueue();
 				r.setBuild(build);
+				if (fromName(r.getName(), stratosResources) !=null){
```

Comment:
If you had a subclass of LockableResource for StratOS objectives then code that updates the objective would belong in that subclass. It doesn't make sense being in the LockableResourceManager class.